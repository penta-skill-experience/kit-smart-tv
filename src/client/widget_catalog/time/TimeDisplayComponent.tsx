import * as React from "react";
import {DisplayComponent} from "../../widget/DisplayComponent";

const weekday = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];

export class TimeDisplayComponent extends DisplayComponent<any> {

    private intervalID: NodeJS.Timer;

    constructor(props) {
        super(props);
        this.state = {
            time: this.getTimeString(),
            date: this.getDateString(),
            day: new Date().getDay()
        };
    }

    componentDidMount() {
        this.intervalID = setInterval(
            () => this.tick(),
            1000
        );
    }

    componentWillUnmount() {
        clearInterval(this.intervalID);
    }

    tick() {
        this.setState({
            time: this.getTimeString(),
            date: this.getDateString(),
            day: new Date().getDay()
        });
    }

    render() {
        return <div className="flex-col">
            <div
                className="font-light sm:text-3xl md:text-4xl lg:text-6xl xl:text-7xl 2xl:text-8xl 8xl:text-10xl sm:text-left sm:-mt-1 md:-mt-2 lg:-mt-3 xl:-mt-5 2xl:-mt-6 4xl:-mt-8 8xl:-mt-12">{this.state.time}</div
            >
            <div
                className="sm:text-xs md:text-sm lg:text-base lg:text-xl 2xl:text-2xl 4xl:text-4xl 8xl:text-5xl sm:text-left sm:-mt-1 md:-mt-2 lg:-mt-3 xl:-mt-5 2xl:-mt-6 4xl:-mt-8 8xl:-mt-12">{weekday[this.state.day]}, {this.state.date}</div>
        </div>
    }

    private getTimeString() {
        return new Date().toLocaleTimeString("de", {hour: '2-digit', minute: '2-digit'});
    }

    private getDateString() {
        return new Date().toLocaleDateString("en", {day: 'numeric', month: 'long'});
    }
}