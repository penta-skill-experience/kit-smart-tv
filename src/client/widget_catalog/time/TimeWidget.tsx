import * as React from "react";
import {Widget} from "../../widget/Widget";
import {WidgetConfigSaver} from "../../widget/WidgetConfigSaver";
import {DigitalTime} from "./DigitalTime";

export class TimeWidget implements Widget {

    createConfigComponent(rawConfig: Object, saver: WidgetConfigSaver): JSX.Element {
        return undefined;
    }

    createDisplayComponent(rawConfig: Object): JSX.Element {
        return <DigitalTime />;
    }

    getTitle(): string {
        return "Time";
    }

    isConfigurable(): boolean {
        return false;
    }

}