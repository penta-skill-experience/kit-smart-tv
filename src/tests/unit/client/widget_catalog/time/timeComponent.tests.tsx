import React from "react";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {configure, shallow} from "enzyme";
import toJson from "enzyme-to-json";
import * as MockDate from "mockdate";
import {TimeDisplayComponent} from "../../../../../client/widget_catalog/time/TimeDisplayComponent";


// This sets the mock adapter on the default instance
configure({adapter: new Adapter()});
describe("time Component Snapshots", () => {
    test("time component Snapshot time", async () => {
        MockDate.set(new Date(2022, 0O2, 0O3, 11, 23, 42, 11));
        jest.useFakeTimers('legacy');
        Promise.resolve().then(() => jest.advanceTimersByTime(15000));
        jest.runOnlyPendingTimers();
        const wrapper = shallow(<TimeDisplayComponent error={(msg => {})} specialBoldFontColor={"ForestGreen"} specialSubtleFontColor={"Tomato"} />);
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();
        jest.useRealTimers();
    });
    afterAll(() => {
        MockDate.reset();
        jest.restoreAllMocks();
    });
});