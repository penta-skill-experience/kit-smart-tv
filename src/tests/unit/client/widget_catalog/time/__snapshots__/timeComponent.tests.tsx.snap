// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time Component Snapshots time component Snapshot time 1`] = `
<div
  className="flex-col"
>
  <div
    className="font-light sm:text-3xl md:text-4xl lg:text-6xl xl:text-7xl 2xl:text-8xl 8xl:text-10xl sm:text-left sm:-mt-1 md:-mt-2 lg:-mt-3 xl:-mt-5 2xl:-mt-6 4xl:-mt-8 8xl:-mt-12"
  >
    11:23
  </div>
  <div
    className="sm:text-xs md:text-sm lg:text-base lg:text-xl 2xl:text-2xl 4xl:text-4xl 8xl:text-5xl sm:text-left sm:-mt-1 md:-mt-2 lg:-mt-3 xl:-mt-5 2xl:-mt-6 4xl:-mt-8 8xl:-mt-12"
  >
    Thursday
    , 
    March 3
  </div>
</div>
`;
